.github/workflows/tests.yml

```diff
@@ -283,9 +300,25 @@ jobs:
         working-directory: frontend/asset-processor
         run: pnpm test
 
-      - name: Plugin QUnit
+      - name: Split core and official plugins
+        if: matrix.build_type == 'system' && (matrix.target == 'core-plugins' || matrix.target == 'official-plugins')
+        run: |
```

Comment:
This is very hard to read. Could we write it in ruby instead? Github allows you to set the 'shell' for run commands. e.g.:

https://github.com/discourse/.github/blob/523c4cce8e3bb62fe6031816919f9b93e3ae4b51/actions/clone-additional-plugins/action.yml#L101